### ISSUE TITLE
Found a WAV file that doesn't play

### ISSUE THREAD

**Issue by Dimev** [Gianclgar/GDScriptAudioImport#3]
As the title suggest: I found a wav file that doesn't play.
I can't send it as it's too large, but the data starts at ~byte 700, and it contains artist names + a junk chunk
changing the loop to go to that causes it to make weird glitchy sound.
I don't really know how to solve this one sadly, but it works fine when imported with the normal audio importer: https://github.com/godotengine/godot/blob/master/editor/import/resource_importer_wav.cpp

Example file: https://mega.nz/file/tFdXlSiL#YWdCsuFSaRtWa8felZX1FWsVer247oAjGyEXPFV34Yk

**Comment by Gianclgar** [comment]
Hi, could you try to send/post the file via a third party service? I recommend https://tmp.ninja/ since it has a 10gb limit.

Looks like an error in the parsing, since it expects the wav files to have the same structure. This could help to adapt the parsing to different pre-data chunk sizes, so it will be very helpful

**Comment by Dimev** [comment]
Ah right I have mega, l'll share it via there
I do think it has to do with the header. I also tried porting the built-in parser to gdscript but that's really slow

**Comment by Dimev** [comment]
It seems to sometimes happen with music downloaded as wav with youtubedl

re-exporting it with audacity fixes the issue

**Comment by pixelriot** [comment]
I've done some testing, and it seems as if wav files with 24 bps don't play properly, while the ones with 16 bps have no problem.
Maybe that helps in finding the bug.
I will continue my research.

**Comment by Gianclgar** [comment]
> I've done some testing, and it seems as if wav files with 24 bps don't play properly, while the ones with 16 bps have no problem.
> Maybe that helps in finding the bug.
> I will continue my research.

It makes sense since the current version doesn't handle bit rate in any way, just prints it. See line 109 etc.

**Comment by Gianclgar** [comment]
> I've done some testing, and it seems as if wav files with 24 bps don't play properly, while the ones with 16 bps have no problem.
> Maybe that helps in finding the bug.
> I will continue my research.

Makes sense since the  bitrate is not handled in the current version, just printed out. See lines 118 and 119. I added a "TODO" in the comments

Also, as mentioned in #7 , the format code "3" , which seems to be for 24bit audio is not handled yet.

**Comment by pixelriot** [comment]
When loading the 24bps wav file directly into a Godot AudioPlayer node everything plays fine, although the stream in GD source code seems to not handle format code "3": https://github.com/godotengine/godot/blob/master/scene/resources/audio_stream_sample.cpp#L368

**Comment by Gianclgar** [comment]
Yup, I've been checking the Godot ResourceImporter code, and it automatically converts bitrates greater than 16bit to 16bit. See lines 289 and below https://github.com/godotengine/godot/blob/a02620f3a5f3b94c68dfbe18874b34ac1c32374a/editor/import/resource_importer_wav.cpp

The file mentioned in #7 is 32bit and when imported with the built-in resource importer it's converted to 16bit.

I assume they did not cover this since they expected users to always use the built-in resource importer

**Comment by Gianclgar** [comment]
This might have been fixed by #13 . Awaiting confirmation!

**Comment by pixelriot** [comment]
> 

Works like a charm. Nearly my whole wav-library can be played back now. Thanks a lot.

However, I found some that don't play. Somehow, the data-part of "those4bytes" seems empty.
https://drive.proton.me/urls/FDQ71VN5WC#zIZqoIAFgZ1n
